### ISSUE TITLE
The installation link in README https://dl.bintray.com/itemis/generic/xtext-testing/updates/ is not valid

### ISSUE THREAD

**Issue by eugenedruy** [itemis/xtext-testing#9]
The link provided in README for installation
https://dl.bintray.com/itemis/generic/xtext-testing/updates/
seems to be not valid anymore
Gives an error
502 Bad Gateway
Please provide a valid link or remove the reference

**Comment by eugenedruy** [comment]
Please note the link on the same server is listed in Franca dependencies for franca-photon
https://github.com/franca/franca/blob/ecb2b98b3a1d0e9b690bd4d454cf5ceb3f584aca/releng/org.franca.targetplatform/franca-photon.target#L59
causing the Franca build failure

**Comment by cdietrich** [comment]
bintray is dead.
have no idea where to host alternatively.
github releases or pages may be a solution, but that woulld need work to figure out how to do that

**Comment by eugenedruy** [comment]
> bintray is dead. have no idea where to host alternatively. github releases or pages may be a solution, but that would need work to figure out how to do that or abuse a second repo to host binaries https://www.lorenzobettini.it/2021/03/publishing-an-eclipse-p2-composite-repository-on-github-pages/

The link from README is probably not such a big issue. However it is the reason why Franca build fails for me on https://github.com/franca/franca/blob/ecb2b98b3a1d0e9b690bd4d454cf5ceb3f584aca/releng/org.franca.targetplatform/franca-photon.target#L59

I've built xtext-testing locally and got it installed into my .m2. However trying to figure out how to fix the Franca build now, no experience with PDEs (.target files)  :)

**Comment by kthoms** [comment]
When you have build it locally you should have the resulting p2 repo below ./releng/eclipse-repository/target.
Best is to put this repository to a location that is reachable for you via HTTP(S).
Then add this repository to your target definition.

**Comment by eugenedruy** [comment]
> When you have build it locally you should have the resulting p2 repo below ./releng/eclipse-repository/target. Best is to put this repository to a location that is reachable for you via HTTP(S). Then add this repository to your target definition.

Thanks @kthoms 
Is there a way just to reference the local artefact without hosting it anywhere with an http access?

**Comment by kthoms** [comment]
Target definitions can also work with absolute file URIs. That makes them system dependent, which is generally bad for build related topics. Better is to serve somewhere. I usually abuse a hosted Nexus repository for that.

**Comment by eugenedruy** [comment]
Thanks Karsten
I've added it for now locally like below (0.2.0.202201291429 is the version created on my machine)
```
    <location includeMode="planner" includeAllPlatforms="false" includeSource="true" includeConfigurePhase="false" type="InstallableUnit">
      <unit id="com.itemis.xtext.testing.feature.feature.group" version="0.2.0.202201291429"/>
      <repository location="file:/my/workspace/path/xtext-testing/releng/eclipse-repository/target/repository" />
<!--
      <unit id="com.itemis.xtext.testing.feature.feature.group" version="0.2.0.201612011358"/>
      <repository location="https://dl.bintray.com/itemis/generic/xtext-testing/releases/0.2.0.201612011358/"/>
-->
    </location>
```

and got an error (see below). I'm a bit confused now whether it indeed resolved the dependency from .target file or it is another missing library (org.eclipse.xtext.testing, which I kind of found here https://mvnrepository.com/artifact/org.eclipse.xtext/org.eclipse.xtext.testing)

```
workspace/franca.main/releng/org.franca.parent$ mvn install
[INFO] Scanning for projects...
[INFO] Computing target platform for MavenProject: org.franca:org.franca.core:0.13.2-SNAPSHOT @ /home/edroi/workspace/franca.main/plugins/org.franca.core/pom.xml
[INFO] Fetching p2.index from http://download.eclipse.org/releases/photon/201806271001/ (142B)
[INFO] Adding repository http://download.eclipse.org/releases/photon/201806271001
[INFO] Fetching p2.index from http://download.eclipse.org/modeling/emf/compare/updates/releases/3.0/R201408191310/ (116B)
[INFO] Fetching p2.index from http://download.eclipse.org/modeling/emf/compare/updates/releases/3.0/R201408191310/ (116B)
[INFO] Adding repository http://download.eclipse.org/modeling/emf/compare/updates/releases/3.0/R201408191310
[INFO] Adding repository http://download.itemis.com/updates/releases/2.1.1
[INFO] Adding repository http://download.eclipse.org/modeling/emft/mwe/updates/releases/2.9.0
[INFO] Fetching p2.index from http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.14.0/ (172B)
[INFO] Fetching p2.index from http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.14.0/ (172B)
[INFO] Adding repository http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.14.0
[INFO] Fetching p2.index from http://rtsys.informatik.uni-kiel.de/~kieler/updatesite/release_pragmatics_2017-02-04/ (117B)
[INFO] Fetching p2.index from http://rtsys.informatik.uni-kiel.de/~kieler/updatesite/release_pragmatics_2017-02-04/ (117B)
[INFO] Adding repository http://rtsys.informatik.uni-kiel.de/~kieler/updatesite/release_pragmatics_2017-02-04
[INFO] Fetching p2.index from https://download.eclipse.org/elk/updates/releases/0.2.3/ (172B)
[INFO] Fetching p2.index from https://download.eclipse.org/elk/updates/releases/0.2.3/ (172B)
[INFO] Adding repository https://download.eclipse.org/elk/updates/releases/0.2.3
[INFO] Adding repository http://kbirken.github.io/dbus-emf-model/releases
[INFO] Adding repository http://franca.github.io/franca/update_site/thirdparty
[INFO] Fetching p2.index from http://franca.github.io/franca/update_site/thirdparty/xtext-testing/0.2.0.202106041111/ (173B)
[INFO] Fetching p2.index from http://franca.github.io/franca/update_site/thirdparty/xtext-testing/0.2.0.202106041111/ (173B)
[INFO] Adding repository file:/home/edroi/workspace/xtext-testing/releng/eclipse-repository/target/repository
[INFO] Adding repository http://meysholdt.github.io/eclipse_jdt_junit_runners/updatesite
[INFO] Adding repository http://www.xpect-tests.org/updatesite/nightly/local
[INFO] Fetching p2.index from http://update.eclemma.org/ (172B)
[INFO] Fetching p2.index from http://update.eclemma.org/ (172B)
[INFO] Adding repository http://update.eclemma.org
[ERROR] Cannot resolve target definition:
[ERROR]   Software being installed: com.itemis.xtext.testing.feature.feature.group 0.2.0.202201291429
[ERROR]   Missing requirement: com.itemis.xtext.testing 0.2.0.202201291429 requires 'bundle org.eclipse.xtext.testing 2.15.0' but it could not be found
[ERROR]   Cannot satisfy dependency: com.itemis.xtext.testing.feature.feature.group 0.2.0.202201291429 depends on: com.itemis.xtext.testing [0.2.0.202201291429]
[ERROR] 
[ERROR] Failed to resolve target definition /home/edroi/workspace/franca.main/releng/org.franca.targetplatform/franca-photon.target: See log for details -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException

```

**Comment by cdietrich** [comment]
you still seem to have 
Fetching p2.index from http://franca.github.io/franca/update_site/thirdparty/xtext-testing/0.2.0.202106041111/ (173B)
[INFO] Fetching p2.index from http://franca.github.io/franca/update_site/thirdparty/xtext-testing/0.2.0.202106041111/ (173B)
[INFO] A
did you created that one?

**Comment by kthoms** [comment]
```
requires 'bundle org.eclipse.xtext.testing 2.15.0' but it could not be found
```
I see that you have Xtext 2.14 in the target. That is too low.
```
Adding repository http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.14.0
```

**Comment by cdietrich** [comment]
there are other places that use 2.14 too here in franca repo.
there are also bugs in xtend-maven-plugin old versions that need workarounds

**Comment by kthoms** [comment]
> you could try to downgrade xtext-testing to use xtext 2.14 but i am not sure if this will work.

IIRC there are reasons for this lower bound and 2.14 might not compatible. I fear that Franca needs an upgrade to a newer version, and this has to be done first before you continue here. You have successfully build this plugin and integrated in your target platform. So for this project you are done. Now you have to make sure that you build Franca with an updated target platform that at least uses Xtext 2.15 (which again is very old).

**Comment by cdietrich** [comment]
no it was just my arbitrariness
https://github.com/itemis/xtext-testing/commit/13864ce3f5bdac67e30ba4277f4407f4321109ce

**Comment by kthoms** [comment]
So we could just lower the requirement here in `releng/target-platform/xtext-testing-tp.target` to 2.14.0?

**Comment by cdietrich** [comment]
yes, and in the manifest, it still builds

**Comment by eugenedruy** [comment]
> 
> also the 202201291429 timestamp does not seem to be freshly built
> 
Just noticed the date on my VM is off. :) Sorry for confusion
workspace/xtext-testing/releng/eclipse-repository/target$ date
Sun Jan 30 07:07:07 EST 2022

**Comment by eugenedruy** [comment]
So I'm a bit confused now which way to go (or in what order)
Should I downgrade xtext-testing dependecy to 2.14 , rebuild it and build franca 
or 
it has to be Xtext 2.15 everywhere?

**Comment by cdietrich** [comment]
you can downgrade to 2.14 (manifest and target)
and then do https://github.com/franca/franca/compare/master...cdietrich:cd_fixBuild2
(with timestamp adjusted)

**Comment by eugenedruy** [comment]
Thanks a lot for your help
I've made the changes mentioned, including the eclemma version change which was not mentioned previously, but I found it out on my own
I've manage to pass the build point I was failing before but hit another build failure and had to open another issue
https://github.com/franca/franca/issues/304
which is not relevant to xtext-testing
To add to what was said I've just noticed the commit mentioned by @cdietrich in the pom file
https://github.com/franca/franca/compare/master...cdietrich:cd_fixBuild2#diff-0cd9e8e35ad0bd9b4506ca4d57820cb1d5ca93d54dd21b2c8ddc14a98db38aff
which haven't been done yet.
Planning to try it a bit later and update on results. Meanwhile it seems like I'm on a deadlock between Java 8 & Java 11
Not sure whether that pom update is supposed to solve it

**Comment by eugenedruy** [comment]
Actually the Franca build passed with @cdietrich 's
https://github.com/franca/franca/compare/master...cdietrich:cd_fixBuild2
commit
Thanks a lot
I'm closing the ticket
https://github.com/franca/franca/issues/304
